### PR TITLE
Updating workshop apps for SDK version 2.4.0

### PR DESCRIPTION
**Pull request by brucearmstrong**
- New SDK version
- Also allowed IDE-prompted Gradle updates.

### CODE REVIEW

**Review by ah1053 — APPROVED** [review]

File: android/WorkshopCompleted/.idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="CompilerConfiguration">
+    <bytecodeTargetLevel target="1.8" />
+  </component>
+</project>
```
Comment (franklin-huang-mobiledgex) [inline android/WorkshopCompleted/.idea/compiler.xml:6]: nit: new line at the end of file to get rid of red circle

**Review by franklin-huang-mobiledgex — APPROVED** [review]
Lgtm!

File: android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java [hunk 2]
```diff
@@ -380,7 +381,7 @@ public boolean findCloudlet() throws ExecutionException, InterruptedException, P
         //Find FqdnPrefix from Port structure.
         String FqdnPrefix = "";
         List<distributed_match_engine.Appcommon.AppPort> ports = mClosestCloudlet.getPortsList();
-        String appPortFormat = "{Protocol: %d, FqdnPrefix: %s, Container Port: %d, External Port: %d, Public Path: '%s'}";
+        String appPortFormat = "{Protocol: %d, Container Port: %d, External Port: %d, Path Prefix: '%s'}";
```
Comment (lgarner) [inline android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java:384]: We got rid of PathPrefix (L7)?

File: android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java [hunk 3]
```diff
@@ -380,7 +381,7 @@ public boolean findCloudlet() throws ExecutionException, InterruptedException, P
...
         for (Appcommon.AppPort aPort : ports) {
             FqdnPrefix = aPort.getFqdnPrefix();
```
Comment (lgarner) [inline android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java:386]: Unused?

Comment (brucearmstrong) [reply android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java:386]: Yes. Just showing how to get the values. (Possibly copied from EmptyMatchingEngineApp.)

**Pull request merged**